packages/vite/src/generators/configuration/configuration.ts

```diff
@@ -28,27 +30,62 @@ export async function viteConfigurationGenerator(tree: Tree, schema: Schema) {
   );
   let buildTargetName = 'build';
   let serveTargetName = 'serve';
+  let testTargetName = 'test';
 
   schema.includeLib ??= projectType === 'library';
 
+  // This is for when we are convering an existing project
+  // to use the vite executors.
   if (!schema.newProject) {
-    const { buildTarget, serveTarget, unsuppored } =
-      findExistingTargets(targets);
+    const {
+      buildTarget,
+      serveTarget,
+      testTarget,
+      unsupported,
+      unsupportedUserProvidedTarget,
+    } = findExistingTargetsInProject(targets, {
+      build: schema.buildTarget,
+      serve: schema.serveTarget,
+      test: schema.testTarget,
+    });
 
     /**
-     * Here, we make sure that the project has a build target
-     * and it is unsupported, before throwing the error.
-     * The reason is that the project could not have a build
-     * target at all, in which case we don't want to throw.
-     * Or the project can have multiple build targets, one of which unsupported.
-     * In that case, we convert the supported one.
-     */
-    if (!buildTarget && unsuppored) {
+     * This means that we only found unsupported build targets in that project.
+     *
+     * If the `unsupported` flag was false, it would mean that we did not find
+     * a build target at all, so we can create a new one.
+     *
+     * So we only throw if we found a target, but it is unsupported.
```

Comment:
This says `if we found a target` but the check is `!buildTarget` so should ie be `if we did not find a target`?

Reply:
Yeah, if the `unsupported` flag is `true`, it means we found a target, but it was unsupported, so we are not returning `buildTarget`. We only set `buildTarget` if it's valid.